core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -628,6 +621,13 @@ private[spark] class ExecutorAllocationManager(
             allocationManager.onExecutorIdle(executorId)
           }
         }
+
+        // If the task is failed, we expect it to be resubmitted later.
+        if (taskEnd.reason != Success) {
+          stageIdToTaskIndices.get(stageId).foreach { taskIndices =>
+            taskIndices.remove(taskIndex)
```

Comment:
as I mentioned earlier, this is insufficient. We need to call `onSchedulerBacklogged` here because we want executors.